### PR TITLE
Add directory service related logs to CloudWatch

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
/var/log/sssd/sssd.log
/var/log/sssd/sssd_default.log
/var/log/parallelcluster/pam_ssh_key_generator.log

This commit also improves `cloudwatch_log_files.json` to allow `dna_key` be an array for nested dicts.

Signed-off-by: Hanwen <hanwenli@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json [hunk 1]
```diff
@@ -265,6 +265,84 @@
...
+            "enabled"
+          ],
+          "satisfying_values": ["true"]
+        }
+      ]
+    },
+    {
+      "timestamp_format_key": "default",
+      "file_path": "/var/log/sssd/sssd_default.log",
+      "log_stream_name": "sssd_default",
```
Comment (tilne) [inline cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json]: default is the name of the domain. can we maybe call this `sssd_default_domain`? or just `sssd_domain`?

File: cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json [hunk 2]
```diff
@@ -265,6 +265,84 @@
...
+      ],
+      "node_roles": [
+        "HeadNode"
+      ],
+      "feature_conditions": [
+        {
+          "dna_key": [
+            "directory_service",
+            "generate_ssh_keys_for_users"
+            ],
```
Comment (tilne) [inline cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json]: formatting

File: cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json (hunk 1 above)
Comment (hanwen-pcluste) [reply cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json]: Done. I changed it to sssd_domain_default to be consistent with what we have in the `sssd.conf` (domain/default)

File: cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json (hunk 2 above)
Comment (hanwen-pcluste) [reply cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/cloudwatch_log_files.json]: Done

**Review by tilne — APPROVED** [review]

**Pull request merged**